.github/workflows/automated-release.yaml

```diff
@@ -0,0 +1,138 @@
+name: Automated Release Operator
+
+on:
+  workflow_dispatch:
+    inputs:
+      gitRef:
+        description: Commit SHA, tag or branch name (usually main branch)
+        required: true
+        default: "main"
+        type: string
+      kuadrantOperatorVersion:
+        description: Kuadrant Operator version (semver)
+        default: 0.0.0
+        type: string
+      authorinoOperatorVersion:
+        description: Authorino Operator bundle version (semver)
+        default: 0.0.0
+        type: string
+      limitadorOperatorVersion:
+        description: Limitador Operator bundle version (semver)
+        default: 0.0.0
+        type: string
+      dnsOperatorVersion:
+        description: DNS Operator bundle version (semver)
+        default: 0.0.0
+        type: string
+      wasmShimVersion:
+        description: WASM Shim version (semver)
+        default: 0.0.0
+        type: string
+      consolePluginVersion:
+        description: ConsolePlugin version (semver)
+        default: 0.0.0
+        type: string
+      olmChannel:
+        description: OLM channel
+        required: false
+        type: string
+        default: "stable"
+
+jobs:
+  auto-release:
+    name: Release operator
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.22.x
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.22.x
+        id: go
+      - name: Checkout code at git ref
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.gitRef }}
+          token: ${{ secrets.KUADRANT_DEV_PAT }}
+
+      - name: Install gettext-base
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gettext-base
+
+      - name: Install yq tool
+        run: |
+          make yq
+
+      - name: Create release branch(es)
+        id: create-release-branch
+        shell: bash
+        run: |
+          base_branch=release-v$(echo "${{ github.event.inputs.kuadrantOperatorVersion }}" | sed 's/-.*//; s/\.[0-9]*$//')
```

Comment:
Looking at this `sed` command I have found it to be fragile, and is easy to break. The test file that I used is 
```
1.0.0
0.0.0
1.0.1
1.0.2
1.1.0
1.2.0
1.0.0-rc
1.0.0-rc1
1.0.0-rc.1
1.0.0+build.1
1.0.0+build
1.0.0+build1
1.0.0-rc+build.1
1.0.0-rc1+build.1
1.0.0-rc.1+build.1
``` 
The ran, where versions.txt is the file above.
```
sed 's/-.*//; s/\.[0-9]*$//' versions.txt 
```
The result from this was.
```
1.0
0.0
1.0
1.0
1.1
1.2
1.0
1.0
1.0
1.0.0+build
1.0.0+build
1.0.0+build1
1.0
1.0
1.0
```
As you can see it does not work when there is a `+build` without `-rc`.

Reply:
buuut, the `+build` is a valid semver tag, so should be there 👍🏼